### Read the current state of the element analysis bcalm table: 
- Expecting: Table of the element based BCalm results using a scrambled control (e.g. "bcalm_adj_p_NGN2") and a column "is_considered_open_acc_SCREEN" which is annotated based on data from e.g. SCREEN information for H1, H9, NPC, neural stem cell and excitatory neuron

In [2]:
import pandas as pd

In [46]:
element_metadata_path = "/home/kisa/coding/80K_MPRA/element_variant_analysis_output/ngn2_wtc11_elements_bcalm_scrambled095_screenv4_annotation_groups_2507_ngn2_neuro_ctrl.tsv.gz"
element_metadata_table = pd.read_csv(element_metadata_path, sep="\t")
element_metadata_table.columns

Index(['name', 'chr', 'start', 'end', 'strand', 'sequence', 'label',
       'raw_activity_NGN2_log2FC_bcalm', 'bcalm_adj_p_NGN2',
       'normalized_activity_NGN2', 'bcalm_data_exists_NGN2',
       'raw_activity_undiffWTC11_log2FC_bcalm', 'bcalm_adj_p_undiffWTC11',
       'normalized_activity_undiffWTC11', 'bcalm_data_exists_undiffWTC11',
       'SCREEN_ID', 'is_NPC_SCREEN_cCRE', 'is_H9_SCREEN_cCRE',
       'is_H1_SCREEN_cCRE', 'is_neural_stem_SCREEN_cCRE',
       'is_excitatory_neuron_SCREEN_cCRE', 'PLS_is_NPC_SCREEN_cCRE',
       'CA-only_is_NPC_SCREEN_cCRE', 'CA-H3K4me3_is_NPC_SCREEN_cCRE',
       'CA-TF_is_NPC_SCREEN_cCRE', 'pELS_is_NPC_SCREEN_cCRE',
       'dELS_is_NPC_SCREEN_cCRE', 'PLS_is_H9_SCREEN_cCRE',
       'CA-only_is_H9_SCREEN_cCRE', 'CA-H3K4me3_is_H9_SCREEN_cCRE',
       'pELS_is_H9_SCREEN_cCRE', 'dELS_is_H9_SCREEN_cCRE',
       'CA-TF_is_H9_SCREEN_cCRE', 'PLS_is_H1_SCREEN_cCRE',
       'CA-TF_is_H1_SCREEN_cCRE', 'CA-H3K4me3_is_H1_SCREEN_cCRE',
       'CA-only_is_H1_SCRE

In [51]:
# annotate the expected column:
columns_of_interest_for_sequences_and_screen = ['name', 'chr', 'start', 'end', 'strand', 'sequence', 'raw_activity_NGN2_log2FC_bcalm', 'bcalm_adj_p_NGN2', 'normalized_activity_NGN2', 'SCREEN_ID', 'is_NPC_SCREEN_cCRE', 'is_H9_SCREEN_cCRE', 'is_H1_SCREEN_cCRE', 'is_neural_stem_SCREEN_cCRE', 'is_excitatory_neuron_SCREEN_cCRE']
element_metadata_table = element_metadata_table[columns_of_interest_for_sequences_and_screen].copy()

element_metadata_table['is_considered_open_acc_SCREEN'] = element_metadata_table[['is_NPC_SCREEN_cCRE', 'is_neural_stem_SCREEN_cCRE', 'is_excitatory_neuron_SCREEN_cCRE']].any(axis=1)
element_metadata_table

,name,chr,start,end,strand,sequence,raw_activity_NGN2_log2FC_bcalm,bcalm_adj_p_NGN2,normalized_activity_NGN2,SCREEN_ID,is_NPC_SCREEN_cCRE,is_H9_SCREEN_cCRE,is_H1_SCREEN_cCRE,is_neural_stem_SCREEN_cCRE,is_excitatory_neuron_SCREEN_cCRE,is_considered_open_acc_SCREEN
0,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70276484.0,70276754.0,-,CTTGAATTTGAATTTAGAGTGAACATGTTGTGATTTTGTATCTGTG...,-0.089726,1.000000e+00,-1.928542,EH38E1825144,False,False,False,False,False,False
1,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70323127.0,70323397.0,-,GCAGTGGCTCATGCCTGCAATCTCAGCACTTCGGGAGGTTGAGGTG...,-0.026591,1.000000e+00,-0.295843,EH38E1825170,False,False,False,False,False,False
2,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70325057.0,70325327.0,-,GCAAGGATCTAAAGTTCCAATTGGCATTAGCTTTCTTTTCGTTTAC...,-0.033114,1.000000e+00,-0.464529,EH38E1825171,False,False,False,False,False,False
3,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70213614.0,70213884.0,-,TTTTTTCAGGAAGTGAAGAATTTAGAGGGATAAAAGGCGGAAATAA...,-0.018290,1.000000e+00,-0.081174,EH38E3188739,False,False,False,False,False,False
4,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70277149.0,70277419.0,-,ATCTGGCAGGCCTTTCCCCTCTGCAGGTGGTGTCACTCGTGGTGGT...,-0.038026,1.000000e+00,-0.591552,EH38E3188758,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
73935,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,106984308.0,106984578.0,+,TAGAAAAGCACAATGAACTGAAACCTTACAAATGCCAGCTCTGCTA...,-0.076844,1.000000e+00,-1.595406,EH38E3899597,False,False,False,False,False,False
73936,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107008116.0,107008386.0,+,AAGCAGAATCCTCCCCACCCCCCTCTCATTATGCAGTTGGGGAGGA...,0.138223,1.641708e-32,3.966275,EH38E3899600,False,False,False,False,False,False
73937,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107019262.0,107019532.0,+,GAAGAGGTCCCCTGGGTCTGAAAATTATAATAGACGGCAGCAGTGA...,0.054619,2.456195e-04,1.804262,EH38E3899604,False,False,False,False,False,False
73938,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107038939.0,107039209.0,+,CTGTCTTCAAGCATCCTAAATCTTTTATATGAGTGTCTCTAGAAGG...,-0.064615,1.000000e+00,-1.279153,EH38E3899614,False,False,False,False,False,False


In [52]:
ref_only_element_metadata_table = element_metadata_table.loc[element_metadata_table['name'].str.contains("cardiac_neuro_cava_random:") & ~element_metadata_table['name'].str.contains(":ALT_")].copy()
ref_only_element_metadata_table['start'] = ref_only_element_metadata_table['start'].astype(int)
ref_only_element_metadata_table['end'] = ref_only_element_metadata_table['end'].astype(int)
ref_only_element_metadata_table

,name,chr,start,end,strand,sequence,raw_activity_NGN2_log2FC_bcalm,bcalm_adj_p_NGN2,normalized_activity_NGN2,SCREEN_ID,is_NPC_SCREEN_cCRE,is_H9_SCREEN_cCRE,is_H1_SCREEN_cCRE,is_neural_stem_SCREEN_cCRE,is_excitatory_neuron_SCREEN_cCRE,is_considered_open_acc_SCREEN
0,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70276484,70276754,-,CTTGAATTTGAATTTAGAGTGAACATGTTGTGATTTTGTATCTGTG...,-0.089726,1.000000e+00,-1.928542,EH38E1825144,False,False,False,False,False,False
1,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70323127,70323397,-,GCAGTGGCTCATGCCTGCAATCTCAGCACTTCGGGAGGTTGAGGTG...,-0.026591,1.000000e+00,-0.295843,EH38E1825170,False,False,False,False,False,False
2,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70325057,70325327,-,GCAAGGATCTAAAGTTCCAATTGGCATTAGCTTTCTTTTCGTTTAC...,-0.033114,1.000000e+00,-0.464529,EH38E1825171,False,False,False,False,False,False
3,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70213614,70213884,-,TTTTTTCAGGAAGTGAAGAATTTAGAGGGATAAAAGGCGGAAATAA...,-0.018290,1.000000e+00,-0.081174,EH38E3188739,False,False,False,False,False,False
4,cardiac_neuro_cava_random:AARS1|ENSG0000009086...,chr16,70277149,70277419,-,ATCTGGCAGGCCTTTCCCCTCTGCAGGTGGTGTCACTCGTGGTGGT...,-0.038026,1.000000e+00,-0.591552,EH38E3188758,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
73935,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,106984308,106984578,+,TAGAAAAGCACAATGAACTGAAACCTTACAAATGCCAGCTCTGCTA...,-0.076844,1.000000e+00,-1.595406,EH38E3899597,False,False,False,False,False,False
73936,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107008116,107008386,+,AAGCAGAATCCTCCCCACCCCCCTCTCATTATGCAGTTGGGGAGGA...,0.138223,1.641708e-32,3.966275,EH38E3899600,False,False,False,False,False,False
73937,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107019262,107019532,+,GAAGAGGTCCCCTGGGTCTGAAAATTATAATAGACGGCAGCAGTGA...,0.054619,2.456195e-04,1.804262,EH38E3899604,False,False,False,False,False,False
73938,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107038939,107039209,+,CTGTCTTCAAGCATCCTAAATCTTTTATATGAGTGTCTCTAGAAGG...,-0.064615,1.000000e+00,-1.279153,EH38E3899614,False,False,False,False,False,False


In [53]:
# add column indicating quantitative readout:
ref_only_element_metadata_table['has_measurements'] = ref_only_element_metadata_table['raw_activity_NGN2_log2FC_bcalm'].notna()
ref_only_element_metadata_table['has_measurements'].value_counts()

# add column indicating significant elements:
ref_only_element_metadata_table['is_significant'] = (ref_only_element_metadata_table['bcalm_adj_p_NGN2'] < 0.05) & ref_only_element_metadata_table['has_measurements']
ref_only_element_metadata_table['is_significant'].value_counts()

is_significant
False    24391
True      3091
Name: count, dtype: int64

In [54]:
ref_only_element_metadata_table_with_measurements = ref_only_element_metadata_table.loc[ref_only_element_metadata_table['has_measurements']].copy()
ref_only_element_metadata_table_with_measurements
print(f"Total elements with measurements: {ref_only_element_metadata_table_with_measurements.shape[0]}")

Total elements with measurements: 25578


### Overview of the elements and their activity in numbers: 
- 25578 total elements
- How many are significantly active? 3091
- How many are significantly more active? 3091
- How many are significantly less active? 0
- How many have no significant activity? 22487

In [55]:
ref_only_element_metadata_table_with_measurements['activity_category'] = ref_only_element_metadata_table_with_measurements.apply(lambda row: 'more_active' if (row['is_significant'] and row['raw_activity_NGN2_log2FC_bcalm'] > 0) else ('less_active' if (row['is_significant'] and row['raw_activity_NGN2_log2FC_bcalm'] < 0) else 'not_significant'), axis=1)
ref_only_element_metadata_table_with_measurements['activity_category'].value_counts()

activity_category
not_significant    22487
more_active         3091
Name: count, dtype: int64

In [56]:
ref_only_element_metadata_table_with_measurements_duplicated = ref_only_element_metadata_table_with_measurements.loc[ref_only_element_metadata_table_with_measurements.duplicated(subset=['chr', 'start', 'end'], keep=False)].sort_values(by=['chr', 'start', 'end'])
ref_only_element_metadata_table_with_measurements_deduplicated = ref_only_element_metadata_table_with_measurements_duplicated.loc[ref_only_element_metadata_table_with_measurements_duplicated['name'].str.contains("_fwd_")].copy()
if ref_only_element_metadata_table_with_measurements_deduplicated.loc[ref_only_element_metadata_table_with_measurements_deduplicated.duplicated(subset=['chr', 'start', 'end'], keep=False)].shape[0] != 0:
    raise ValueError("There are still duplicated elements after filtering for _fwd_ names.")
else:
    print("No duplicated elements remain after filtering for _fwd_ names.")

# combine the deduplicated elements with the non-duplicated elements:
ref_only_element_metadata_table_with_measurements_without_duplicates = ref_only_element_metadata_table_with_measurements.loc[~ref_only_element_metadata_table_with_measurements.duplicated(subset=['chr', 'start', 'end'], keep=False)].copy()
ref_only_element_metadata_table_with_measurements_no_duplicates = pd.concat([ref_only_element_metadata_table_with_measurements_deduplicated, ref_only_element_metadata_table_with_measurements_without_duplicates], ignore_index=True)
ref_only_element_metadata_table_with_measurements_no_duplicates

No duplicated elements remain after filtering for _fwd_ names.


,name,chr,start,end,strand,sequence,raw_activity_NGN2_log2FC_bcalm,bcalm_adj_p_NGN2,normalized_activity_NGN2,SCREEN_ID,is_NPC_SCREEN_cCRE,is_H9_SCREEN_cCRE,is_H1_SCREEN_cCRE,is_neural_stem_SCREEN_cCRE,is_excitatory_neuron_SCREEN_cCRE,is_considered_open_acc_SCREEN,has_measurements,is_significant,activity_category
0,cardiac_neuro_cava_random:REF_DRD4|ENSG0000006...,chr11,596434,596704,+,TCTGGGAAGGGGGCCCCACAGGCAGCACCTACCGCAGGAGCTCAGT...,-0.025263,1.000000e+00,-0.261505,EH38E2937745,False,False,False,False,False,False,True,False,not_significant
1,cardiac_neuro_cava_random:REF_DRD4|ENSG0000006...,chr11,606724,606994,+,TTTTCTGCATTGATTTCCCTTCTTGAGCAGTTTCTCAGTTAGCTCC...,-0.026812,1.000000e+00,-0.301550,EH38E2937757,False,False,False,False,False,False,True,False,not_significant
2,cardiac_neuro_cava_random:DRD4|ENSG00000069696...,chr11,611217,611487,+,GTATTGCCACATCCTGTAGGCCTGGGGTCAAGCCTGTTCGCCTGTG...,-0.024900,1.000000e+00,-0.252123,EH38E2937769,False,False,False,False,False,False,True,False,not_significant
3,cardiac_neuro_cava_random:REF_DRD4|ENSG0000006...,chr11,612960,613230,+,GAGCACATGGCCTCCCCTCCTTGAGGCTCTGGTCTCACCTTCACCA...,-0.048680,1.000000e+00,-0.867083,EH38E2937773,False,False,False,False,False,False,True,False,not_significant
4,cardiac_neuro_cava_random:DRD4|ENSG00000069696...,chr11,616626,616896,+,ACAAGCCGCAAGGGATCCCTTGATGGTTCTGGGCATGGGCGCCCAG...,0.000257,1.000000e+00,0.398439,EH38E2937784,False,False,False,False,False,False,True,False,not_significant
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
25380,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,106984308,106984578,+,TAGAAAAGCACAATGAACTGAAACCTTACAAATGCCAGCTCTGCTA...,-0.076844,1.000000e+00,-1.595406,EH38E3899597,False,False,False,False,False,False,True,False,not_significant
25381,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107008116,107008386,+,AAGCAGAATCCTCCCCACCCCCCTCTCATTATGCAGTTGGGGAGGA...,0.138223,1.641708e-32,3.966275,EH38E3899600,False,False,False,False,False,False,True,True,more_active
25382,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107019262,107019532,+,GAAGAGGTCCCCTGGGTCTGAAAATTATAATAGACGGCAGCAGTGA...,0.054619,2.456195e-04,1.804262,EH38E3899604,False,False,False,False,False,False,True,True,more_active
25383,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107038939,107039209,+,CTGTCTTCAAGCATCCTAAATCTTTTATATGAGTGTCTCTAGAAGG...,-0.064615,1.000000e+00,-1.279153,EH38E3899614,False,False,False,False,False,False,True,False,not_significant


In [57]:
# filter for active vs non active elements and write the bed files and print the number of elements in each category:
elements_not_significant = ref_only_element_metadata_table_with_measurements_no_duplicates.loc[~ref_only_element_metadata_table_with_measurements_no_duplicates['is_significant']].copy()
print(f"Number of non-significantly active elements: {elements_not_significant.shape[0]}")
more_active_elements = ref_only_element_metadata_table_with_measurements_no_duplicates.loc[ref_only_element_metadata_table_with_measurements_no_duplicates['is_significant'] & (ref_only_element_metadata_table_with_measurements_no_duplicates['raw_activity_NGN2_log2FC_bcalm'] > 0)].copy()
print(f"Number of significantly more active elements: {more_active_elements.shape[0]}")

less_active_elements = ref_only_element_metadata_table_with_measurements_no_duplicates.loc[ref_only_element_metadata_table_with_measurements_no_duplicates['is_significant'] & (ref_only_element_metadata_table_with_measurements_no_duplicates['raw_activity_NGN2_log2FC_bcalm'] < 0)].copy()
print(f"Number of significantly less active elements: {less_active_elements.shape[0]}")

active_elements = pd.concat([more_active_elements, less_active_elements], ignore_index=True)
print(f"Total number of significantly active elements: {active_elements.shape[0]}")

Number of non-significantly active elements: 22307
Number of significantly more active elements: 3078
Number of significantly less active elements: 0
Total number of significantly active elements: 3078


In [58]:
active_elements

,name,chr,start,end,strand,sequence,raw_activity_NGN2_log2FC_bcalm,bcalm_adj_p_NGN2,normalized_activity_NGN2,SCREEN_ID,is_NPC_SCREEN_cCRE,is_H9_SCREEN_cCRE,is_H1_SCREEN_cCRE,is_neural_stem_SCREEN_cCRE,is_excitatory_neuron_SCREEN_cCRE,is_considered_open_acc_SCREEN,has_measurements,is_significant,activity_category
0,cardiac_neuro_cava_random:REF_DRD4|ENSG0000006...,chr11,658260,658530,+,TACTAAAAATACAAAAAAATTAGCCAGGCGTGTTGGCGGGTGCCTG...,0.033139,1.917734e-02,1.248799,EH38E2937859,False,False,False,False,False,False,True,True,more_active
1,cardiac_neuro_cava_random:DRD4|ENSG00000069696...,chr11,658697,658967,+,AGGAGACTGCAGCAGAGCACCCTGCGGGGAGTCCCGCCCAAGCAGC...,0.060959,1.815290e-05,1.968217,EH38E2937860,False,False,False,False,False,False,True,True,more_active
2,cardiac_neuro_cava_random:DRD4|ENSG00000069696...,chr11,659652,659922,+,GTTGGCCTCGGGGTTCAGAGCCGCTGGGAGAACCCTGAGTAGCCTT...,0.106607,3.229130e-03,3.148683,EH38E2937862,False,False,False,False,False,False,True,True,more_active
3,cardiac_neuro_cava_random:PTPN11|ENSG000001792...,chr12,112386316,112386586,+,GATCTAGGGCTTAAAGTCTTTAGACTGCATAATGAATAGTTATTTA...,0.163837,1.364456e-33,4.628677,EH38E3041811,False,False,False,False,False,False,True,True,more_active
4,cardiac_neuro_cava_random:REF_TSC2|ENSG0000010...,chr16,2095960,2096230,+,TGATGGTTATCTGCCTTTTGTAAATCAGGAAGTTTTCCTTTTTGTT...,0.026642,3.294165e-02,1.080769,EH38E3162829,False,False,False,False,False,False,True,True,more_active
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3073,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,106929857,106930127,+,AGTGACTTAGCTTGCCAGAGAGCTCAATAAGTCAATTAAACATTTC...,0.057748,2.966758e-04,1.885172,EH38E3899564,False,False,False,False,False,False,True,True,more_active
3074,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,106955074,106955344,+,GGGCATTTGTTTGATTGTTTGAGCACTCACTCATCTTAGTGTGTGT...,0.039619,3.913232e-06,1.416357,EH38E3899579,False,False,False,False,False,False,True,True,more_active
3075,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,106957994,106958264,+,TACTGAGTTCAATAGAAACAGTTAGATGAACAGTTCAGATGAACAA...,0.023174,7.120797e-03,0.991101,EH38E3899583,False,False,False,False,False,False,True,True,more_active
3076,cardiac_neuro_cava_random:ZNF462|ENSG000001481...,chr9,107008116,107008386,+,AAGCAGAATCCTCCCCACCCCCCTCTCATTATGCAGTTGGGGAGGA...,0.138223,1.641708e-32,3.966275,EH38E3899600,False,False,False,False,False,False,True,True,more_active


In [59]:
# are there any duplicates:
if active_elements.loc[active_elements.duplicated(subset=['chr', 'start', 'end'], keep=False)].shape[0] != 0:
    raise ValueError("There are still duplicated elements in the active elements.")
else:
    print("No duplicated elements remain in the active elements.")

if elements_not_significant.loc[elements_not_significant.duplicated(subset=['chr', 'start', 'end'], keep=False)].shape[0] != 0:
    raise ValueError("There are still duplicated elements in the non-significantly active elements.")
else:
    print("No duplicated elements remain in the non-significantly active elements.")

No duplicated elements remain in the active elements.
No duplicated elements remain in the non-significantly active elements.


In [66]:
import os
output_directory = "/home/kisa/coding/80K_MPRA/80K-Analysis/06_variant_element_analysis/smk_element_driver_enrichment/resources"
output_name_active = "NGN2_scramble_ctrl_active_elements.bed.gz"
output_name_inactive = "NGN2_scramble_ctrl_inactive_elements.bed.gz"
writing=False
if writing:
    active_elements[['chr', 'start', 'end', 'name']].to_csv(os.path.join(output_directory, output_name_active), sep="\t", header=False, index=False, compression="gzip")
    elements_not_significant[['chr', 'start', 'end', 'name']].to_csv(os.path.join(output_directory, output_name_inactive), sep="\t", header=False, index=False, compression="gzip")

#### Between considered open and closed active sequences:

In [63]:
active_considered_open_acc_SCREEN = active_elements.loc[active_elements['is_considered_open_acc_SCREEN']].copy()
print(f"Number of active elements that are considered open in SCREEN: {active_considered_open_acc_SCREEN.shape[0]} ({(active_considered_open_acc_SCREEN.shape[0] / active_elements.shape[0]) * 100:.2f}%)")
active_considered_closed_acc_SCREEN = active_elements.loc[~active_elements['is_considered_open_acc_SCREEN']].copy()
print(f"Number of active elements that are considered closed in SCREEN: {active_considered_closed_acc_SCREEN.shape[0]} ({(active_considered_closed_acc_SCREEN.shape[0] / active_elements.shape[0]) * 100:.2f}%)")


Number of active elements that are considered open in SCREEN: 327 (10.62%)
Number of active elements that are considered closed in SCREEN: 2751 (89.38%)


In [ ]:
output_name_active_open = "NGN2_scramble_ctrl_active_elements_considered_open_acc_SCREEN.bed.gz"
output_name_active_closed = "NGN2_scramble_ctrl_active_elements_considered_closed_acc_SCREEN.bed.gz"
writing=False
if writing:
    active_considered_open_acc_SCREEN[['chr', 'start', 'end', 'name']].to_csv(os.path.join(output_directory, output_name_active_open), sep="\t", header=False, index=False, compression="gzip")
    active_considered_closed_acc_SCREEN[['chr', 'start', 'end', 'name']].to_csv(os.path.join(output_directory, output_name_active_closed), sep="\t", header=False, index=False, compression="gzip")